src/lib/log.c

```diff
@@ -57,12 +66,22 @@ static void log_internal(FILE *out, const char *file, int line, const char *labe
 		if (printed < DEBUG_HEADER_SIZE) {
 			fprintf(out, "%*c", 50 - printed, ' ');
 		}
+
+		/* all debug messages need to finish with a line break, even if the
+		 * user didn't add one to the message, otherwise the output will be
+		 * messed up */
+		if (format[strlen(format) - 1] != '\n') {
+			format_end = "\n";
+		}
 	}
 
 	if (label) {
 		fprintf(out, "%s: ", label);
 	}
 	vfprintf(out, format, args_list);
+	fprintf(out, "%s", format_end);
```

Comment:
I like the idea, but I'd move the if here so you won't need to have another variable for that.

Reply:
Well the if cannot be moved because the time(file:line) has to be printed before the message. I thought of adding another if (cur_log_level == LOG_DEBUG) {} here but using a variable seemed cleaner. But I can use another if instead if you think is better.

Reply:
I prefer to have another if. And then we won't call fprintf again.